src/pal/src/misc/perftrace.cpp

```diff
@@ -659,26 +659,29 @@ PERFlushAllLogs( )
 
 static
 void
-PERFLogFileName(PathCharString * destFileString, const char *fileName, const char *suffix, int max_length)
+PERFLogFileName(PathCharString& destFileString, const char *fileName, const char *suffix)
 {
     const char *dir_path;
     CPalThread* pThread = InternalGetCurrentThread();
-    char * destFileName = (char*)InternalMalloc(pThread, max_length);
     dir_path = (profile_log_path == NULL) ? "." : profile_log_path;
-
+        
+    destFileString.Append(dir_path, strlen(dir_path));
+    destFileString.Append(PATH_SEPARATOR, strlen(PATH_SEPARATOR));
     if (fileName != NULL)
     {
-        snprintf(destFileName, max_length, "%s%s%s", dir_path, PATH_SEPARATOR, fileName);
+        destFileString.Append(fileName, strlen(fileName));
     }
     else
     {
-        snprintf(destFileName, max_length, "%s%s%d_%d%s", dir_path, PATH_SEPARATOR,
-            program_info.process_id, THREADSilentGetCurrentThreadId(), suffix);
+        std::string process_id     = std::to_string(program_info.process_id);
```

Comment:
Not in this case, right. But if the filename in the "if" branch is longer than the little extra and the dir_path didn't fit the internal buffer, you'd get an extra allocation for the Append(filename). That was my point about not adding the Append method - you cannot be sure whether it will cause additional allocations or not in general. While in the case you estimate the total size and then do sprintf / strcat into the buffer, you can be sure that you get at most one allocation. And allocations are costy.